provisioning/service/src/Config/IndividualEnrollment.cs

```diff
@@ -250,17 +254,25 @@ public string DeviceId
         /// Attestation Mechanism
         /// </summary>
         [JsonProperty(PropertyName = "attestation")]
-        private AttestationMechanism _Attestation;
+        private AttestationMechanism _attestation;
         [JsonIgnore]
         public Attestation Attestation
         {
             get
             {
-                return _Attestation.GetAttestation();
+                return _attestation.GetAttestation();
             }
             set
             {
-                _Attestation = new AttestationMechanism(value ?? throw new ArgumentException("Attestation cannot be null."));
+                if (value is X509Attestation)
+                {
+                    if ((((X509Attestation)value ?? throw new ArgumentException("Attestation cannot be null.")).ClientCertificates == null) &&
```

Comment:
null checking should be done before usage. Also the above statement is very hard to read.

ArgumentNullException should be thrown instead.

Reply:
C# tests ?? before use the value. Agreed with the ArgumentNullException.